pkg/cloudprovider/providers/openstack/openstack_volumes.go

```diff
@@ -23,20 +23,181 @@ import (
 	"path"
 	"strings"
 
-	"k8s.io/kubernetes/pkg/volume"
+	k8s_volume "k8s.io/kubernetes/pkg/volume"
 
-	"github.com/rackspace/gophercloud"
-	"github.com/rackspace/gophercloud/openstack"
-	"github.com/rackspace/gophercloud/openstack/blockstorage/v1/volumes"
-	"github.com/rackspace/gophercloud/openstack/compute/v2/extensions/volumeattach"
-	"github.com/rackspace/gophercloud/pagination"
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack"
+	volumes_v1 "github.com/gophercloud/gophercloud/openstack/blockstorage/v1/volumes"
+	volumes_v2 "github.com/gophercloud/gophercloud/openstack/blockstorage/v2/volumes"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/extensions/volumeattach"
+	"github.com/gophercloud/gophercloud/pagination"
 
 	"github.com/golang/glog"
 )
 
+type IVolumes interface {
```

Comment:
Naming is hard; however, you may want to consider changing the name of this interface to be more idiomatic (maybe `VolumeService` with `CreateVolume` and `DeleteVolume`).